Add `deny-all` network policy to `Shoot`'s `kube-system` namespace (#11502)
gardener/gardener: docs/usage/shoot/shoot_kubernetes_versions.md

> ---

Reviewer: Add to this to the index (docs/README.md).

title: Shoot cluster supported Kubernetes versions and specifics
description: Defining the differences and requirements for updating to a supported Kubernetes version
---

> # Shoot Kubernetes version update

Reviewer: ```suggestion
# Shoot Kubernetes Minor Version Upgrades
```

> Breaking changes may be introduced with new Kubernetes versions. This documentation describes the differences and requirements for updating to a supported Kubernetes version.

Reviewer: ```suggestion
Breaking changes may be introduced with new Kubernetes versions.
This documentation describes the differences and requirements for updating to a supported Kubernetes version.
```

one line per sentence

> ## Updating to Kubernetes `v1.33`

Reviewer: ```suggestion
## Upgrading to Kubernetes `v1.33`
```

> ## Updating to Kubernetes `v1.33`

Reviewer: Could you also list the changes for the other versions, please, now that you touch this doc anyways? I don't think that's too much. The changes/requirements can be derived from https://github.com/gardener/gardener/blob/master/pkg/apis/core/validation/shoot.go and https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/shoot/maintenance/reconciler.go, I guess.

> When updating to Kubernetes version `v1.33` a new `deny-all` `NetworkPolicy` is deployed in the `kube-system` namespace of the `Shoot` cluster. For `Shoot` owners that have workloads in the `kube-system` namespace it would be needed from them to specify the expected `Ingress` and `Egress` traffic in `kube-system` `NetworkPolicies`.

Reviewer: ```suggestion
- A new `deny-all` `NetworkPolicy` is deployed into the `kube-system` namespace of the `Shoot` cluster. `Shoot` owners that run workloads in the `kube-system` namespace are required to explicitly allow their  expected `Ingress` and `Egress` traffic in `kube-system` via `NetworkPolicies`.
```

## Upgrading to Kubernetes `v1.33`

- A new `deny-all` `NetworkPolicy` is deployed into the `kube-system` namespace of the `Shoot` cluster. `Shoot` owners that run workloads in the `kube-system` namespace are required to explicitly allow their expected `Ingress` and `Egress` traffic in `kube-system` via `NetworkPolicies`.

## Upgrading to Kubernetes `v1.32`

- The `Shoot`'s field `spec.kubernetes.kubeAPIServer.oidcConfig` is forbidden. `Shoot` owners that have used `oidcConfig` are recommended to configure `StructuredAuthentication`. More information about `StructuredAuthentication` can be found [here](./shoot_access.md#structured-authentication)

## Upgrading to Kubernetes `v1.31`

- The `Shoot`'s field `spec.kubernetes.kubeAPIServer.oidcConfig.clientAuthentication` is forbidden.
- The `Shoot`'s fields `.spec.kubernetes.kubelet.systemReserved` and `.spec.provider.workers[].kubernetes.kubelet.systemReserved` are forbidden. `Shoot` owners should use the `.spec.kubernetes.kubelet.kubeReserved` and `.spec.provider.workers[].kubernetes.kubelet.kubeReserved` fields.

## Upgrading to Kubernetes `v1.30`

> - The `kubelet` `UnlimitedSwap` behavior, configured in the `Shoot`'s and `Worker`'s field `kubernetes.kubelet.memorySwap.swapBehavior`, can no longer be used.

Reviewer: ```suggestion
- The `kubelet` `UnlimitedSwap` behavior, configured in the `Shoot`'s `.spec.{kubernetes,provider.workers[]}.kubelet.memorySwap.swapBehavior` fields, can no longer be used.
```
